### ISSUE TITLE
node_modules/@types/restify/index.d.ts(14,24): error TS2307: Cannot find module 'http2'

### ISSUE THREAD

**Issue by XeniaSiskaki** [DefinitelyTyped/DefinitelyTyped#26260]
I'm using in my project `@types/swagger-express-mw@0.7.0` which has a nested dependency on `restify`. A few months ago `restify` added [HTTP/2 support](https://github.com/restify/node-restify/commit/6b202853d62394f0448486c9b5bbc18589fd44e2) which is only available from node8.4.0. The problem is that although in the library they check [whether `http2` is available](https://github.com/restify/node-restify/commit/6b202853d62394f0448486c9b5bbc18589fd44e2#diff-c945a46d13b34fcaff544d966cffcabaR34), the [latest types update](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/26152/commits/5de7fb22740c2a0e320239a5adfc93cc65a8f0d6#diff-3b16a0ff2be2d161e398cae98da62f21R11) assumes the `http2` module is installed and thus breaking projects that use node<8.4.0 when compiling with the error `node_modules/@types/restify/index.d.ts(14,24): error TS2307: Cannot find module 'http2'`. 

I'm using `typescript@2.3.4` and `node@6.11.1`

cc: @blittle @stevehipwell @leanazulyoro @mgebundy

**Comment by mgebundy** [comment]
Hi @XeniaSiskaki, I'm looking into this.

**Comment by XeniaSiskaki** [comment]
Sorry for the misunderstanding! By nested dependency I meant that `swagger-express-mw` depends on another module which itself installs `restify`.

Doing `npm ls` shows:
```
├─┬ @types/swagger-express-mw@0.7.6
│ └─┬ @types/swagger-node-runner@0.5.9
│   ├── @types/events@1.2.0 deduped
│   ├── @types/express@4.11.1 deduped
│   ├─┬ @types/hapi@16.1.14
│   │ ├── @types/boom@4.3.10
│   │ ├─┬ @types/catbox@7.1.6
│   │ │ └── @types/boom@4.3.10 deduped
│   │ ├── @types/events@1.2.0 deduped
│   │ ├── @types/joi@13.0.8
│   │ ├─┬ @types/mimos@3.0.1
│   │ │ └── @types/mime-db@1.27.0
│   │ ├── @types/node@6.0.112 deduped
│   │ ├── @types/podium@1.0.0
│   │ └─┬ @types/shot@3.4.0
│   │   └── @types/node@6.0.112 deduped
│   ├── @types/node@6.0.112 deduped
│   ├─┬ @types/restify@7.2.0  <--------------- here
│   │ ├─┬ @types/bunyan@1.8.4
│   │ │ ├── @types/events@1.2.0 deduped
│   │ │ └── @types/node@6.0.112 deduped
│   │ ├── @types/node@6.0.112 deduped
│   │ └─┬ @types/spdy@3.4.4
│   │   └── @types/node@6.0.112 deduped
│   └── @types/swagger-schema-official@2.0.10
```
Hope this helps!

**Comment by mgebundy** [comment]
@XeniaSiskaki - My PR has been merged, version 7.2.1 of `@types/restify` should work for you.

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
